### ISSUE TITLE
keepassxc icon shows in entry fields even if that URL is not in keepassxc database

### ISSUE THREAD

**Issue by freedriven** [keepassxreboot/keepassxc-browser#963]
keepassxc puts an icon in every username and password field on every web site
I have 40 or so URL's saved in keepassxc, but don't care to put any and all of my passwords in keepassxc just so I don't have to remember which URLs are in keepassxc or not. As is, I'm compelled to click on the icon, only to (sometimes, or often) find out my password isn't in keepassxc.

## Expected Behavior

keepassxc should not put an icon in a field for which no URL is listed in its database

## Possible Solution

Would be great to have an option in settings to implement the above

## Steps to Reproduce (for bugs)

1.
2.
3.
4.

## Debug info

KeePassXC - {VERSION}
KeePassXC-Browser - {VERSION}
Operating system: Mac/Win/Linux
Browser: Chrome/Firefox/Vivaldi/Chromium

**Comment by varjolintu** [comment]
FYI: You can disable the password icons from the settings.

**Comment by varjolintu** [comment]
You can disable the icons in the settings if you don't like to use them. The username icon is also used to unlock the database and fill credentials right away, so showing them is the default behavior.

**Comment by freedriven** [comment]
Yes, but my point is to use keepassxc to inform me which passwords I have in the database, and which I don't, so I'm not clicking the icon and getting "no password found". Disabling all icons doesn't help with that.

**Comment by varjolintu** [comment]
Currently the extension doesn't work that way. This is why "feature request" tag is added for just showing the icons when you have a credentials for a certain site.

**Comment by freedriven** [comment]
Yea, thanks.  I intended it to be a feature request. Thanks!

**Comment by jabrugger** [comment]
Yes, please, "fix" this. It's useless to have an indicator of "entry available" when there is none available. 
Thanks

**Comment by varjolintu** [comment]
Just to clarify, the icon itself is not an indicator of "entry available". It provides an access to multiple functions, for example unlocking the database when it's locked, and mostly it's a quick way to tell if the extension is connected, database is locked/unlocked etc. Basically it has the same function as the extension icon at the toolbar. The extension icon is also still visible even if you don't have any entries for a site.

**Comment by jabrugger** [comment]
> Just to clarify, the icon itself is not an indicator of "entry available". It provides an access to multiple functions, for example unlocking the database when it's locked, and mostly it's a quick way to tell if the extension is connected, database is locked/unlocked etc. Basically it has the same function as the extension icon at the toolbar. The extension icon is also still visible even if you don't have any entries for a site.

Yes, I know that. And it's what you think the user "should" think. But from an UX point of view, an icon on the username or password field means "here is something you can do with this icon AND these fields".  For general info (DB open, closed, etc) the  icon in the browser bar is ok (and only there, for that kind of purposes). 
Thanks

**Comment by droidmonkey** [comment]
Perhaps we can show the icon with a red X or similar to indicate that we are connected to the database but no entries were found for the site.

**Comment by varjolintu** [comment]
Or, if we want the icon behave identically with the toolbar icon, it has a question mark when credentials are available.

**Comment by droidmonkey** [comment]
we can try it

**Comment by Chealer** [comment]
> You can disable the icons in the settings if you don't like to use them. The username icon is also used to unlock the database and fill credentials right away, so showing them is the default behavior.

I believe this happens even when the database is unlocked.

@freedriven: The Debug info section in the template is supposed to be modified to describe the system on which you experienced the issue.

**Comment by Chealer** [comment]
> Or, if we want the icon behave identically with the toolbar icon, it has a question mark when credentials are available.



> Perhaps we can show the icon with a red X or similar to indicate that we are connected to the database but no entries were found for the site.

[A red X indicates that KeePassXC is not running or is disconnected](https://keepassxc.org/docs/KeePassXC_GettingStarted#_using_the_browser_extension).

Nevertheless, I do think showing something (different) would be best, to distinguish from KeePassXC-Browser just being disabled/broken, not yet loaded, or unable to detect the login form (for example).

**Comment by Chealer** [comment]
> Or, if we want the icon behave identically with the toolbar icon, it has a question mark when credentials are available.

What❓Does this behavior persist in the current KeePassXC-Browser version?

**Comment by varjolintu** [comment]
> > Or, if we want the icon behave identically with the toolbar icon, it has a question mark when credentials are available.
> 
> What❓Does this behavior persist in the current KeePassXC-Browser version?

No. There's no question mark icon in the extension anymore. Instead the icon has a number of credentials found for the site.

**Comment by Chealer** [comment]
Phew
I would agree with displaying "0" or the null sign in these cases.

Can someone with the necessary permission retitle this back as an issue report (rather than a feature request), and more functionally? I suggest:

> Sites without credentials in DB undistinguishable (identical KeePassXC icon shown in form fields)

**Comment by varjolintu** [comment]
> Phew I would agree with displaying "0" or the null sign in these cases.
> 
> Can someone with the necessary permission retitle this back as an issue report (rather than a feature request), and more functionally? I suggest:
> 
> > Sites without credentials in DB undistinguishable (identical KeePassXC icon shown in form fields)

No, because this is a feature request. If there's a separate bug, a new issues is needed. I don't want to track multiple features and issues in a single thread.

**Comment by varjolintu** [comment]
> Sorry @varjolintu, my question was rhetorical. This ticket is an issue report (not a feature request), and should be titled as such (describing the issue rather than any possible solution).
> 
> What I meant to ask was for a developer whose account has sufficient permissions to retitle this ticket to please restore a proper title.

See https://github.com/keepassxreboot/keepassxc-browser/issues/963#issuecomment-667990297

**Comment by Chealer** [comment]
> > Sorry @varjolintu, my question was rhetorical. This ticket is an issue report (not a feature request), and should be titled as such (describing the issue rather than any possible solution).
> > What I meant to ask was for a developer whose account has sufficient permissions to retitle this ticket to please restore a proper title.
> 
> See [#963 (comment)](https://github.com/keepassxreboot/keepassxc-browser/issues/963#issuecomment-667990297)

What is your point?

 
> In short, the extension needs to replace the on/off setting with:
> 
>     1. Display icons
> 
>     2. Don't display icons
> 
>     3. Display icons only if credentials are available

You mean with a setting which offers these 3 options (instead of just #1 and #2)?

**Comment by varjolintu** [comment]
> You mean with a setting which offers these 3 options (instead of just #1 and #2)?

Yes.

**Comment by Chealer** [comment]
That was one **possible** "solution", which:

1. is not ideal in terms of results, as explained above,
2. has no effect until extra steps are taken (either user intervention or making default and possibly drop current config)
3. requires work from extension translators

There was already discussion about a proper solution.

**Comment by varjolintu** [comment]
> That was one **possible** "solution", which:
> 
>     1. is not ideal in terms of results, as explained above,
> 
>     2. has no effect until extra steps are taken (either user intervention or making default and possibly drop current config)
> 
>     3. requires work from extension translators
> 
> 
> There was already discussion about a proper solution.

The proper solution is the one I already described.
Changing default settings always needs user interaction.
Translation work is always needed.

**Comment by Chealer** [comment]
> > That was one **possible** "solution", which:
> > ```
> > 1. is not ideal in terms of results, as explained above,
> > 
> > 2. has no effect until extra steps are taken (either user intervention or making default and possibly drop current config)
> > 
> > 3. requires work from extension translators
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > There was already discussion about a proper solution.
> 
> The proper solution is the one I already described.

If you're referring to something like what @droidmonkey proposed, that's also what I was referring to.


> Changing default settings always needs user interaction.

No

> Translation work is always needed.

If you mean translations will never be over, that's obviously true, but that doesn't mean we can't take care to reduce the burden. What I meant is that it wouldn't be an optimal use of our resources.

**Comment by varjolintu** [comment]
> If you're referring to something like what @droidmonkey proposed, that's also what I was referring to.

That comment is from four years ago, and we didn't have any icon states until later. Currently the icon is gray if no credentials are not found, and with X if connection to KeePassXC has not been made.

The proposal is to *hide* the icon totally when there are no credentials, instead of the gray icon.

> No

Of course it does. We don't want to introduce this as a new default.
 
> If you mean translations will never be over, that's obviously true, but that doesn't mean we can't take care to reduce the burden. What I meant is that it wouldn't be an optimal use of our resources.

New settings need new translations. That's how it goes. Most of the burder is on us, not the translators.

**Comment by varjolintu** [comment]
I'm gonna close the conversation for a while because this is not helping the issue at all.
